core/meta/src/TClass.cxx

```diff
@@ -3451,38 +3457,31 @@ const char *TClass::GetSharedLibs()
 
 TList *TClass::GetListOfBases()
 {
-   if (!fBase) {
-      if (fCanLoadClassInfo) {
-         if (fState == kHasTClassInit) {
+   if (fBase)
+      return fBase;
 
-            R__LOCKGUARD(gInterpreterMutex);
-            // NOTE: Add test to prevent redo if another thread has already done the work.
-            // if (!fHasRootPcmInfo) {
+   if (fState == kHasTClassInit) {
+      R__LOCKGUARD(gInterpreterMutex);
 
-            // The bases are in our ProtoClass; we don't need the class info.
-            TProtoClass *proto = TClassTable::GetProtoNorm(GetName());
-            if (proto && proto->FillTClass(this)) {
-               // Not sure this code is still needed
```

Comment:
You removed this note, are you sure it is not needed?  I.e. is there a actual use case where this branch is taken (obviously I was not sure and needed some more time to investigate)